crowbar_framework/app/models/node_object.rb

```diff
@@ -1064,6 +1064,43 @@ def ssh_cmd(cmd)
     [200, {}]
   end
 
+  # Check for the presence of given file on the node
+  def file_exists?(file)
+    out = run_ssh_cmd("test -e #{file}")
+    out[:exit_code].zero?
+  end
+
+  # Executes a script in background and Waits until it finishes.
+  # We expect that the script generates two kinds of files to indicate success or failure.
+  # Raise a timeout exception if the waiting time exceedes 'seconds'
+  def wait_for_script_to_finish(script, seconds)
+    ssh_status = ssh_cmd(script)
+    if ssh_status[0] != 200
+      raise "Executing of script #{script} has failed on node #{@node.name}."
+    end
+
+    base = script.split(".sh").first
```

Comment:
IIUC `script` is a full path, so this will mean `base` begins with something like `/usr/sbin/...` which is wrong.  That's why I suggested using `File.basename`.